src/runmode-dpdk.c

```diff
@@ -0,0 +1,1295 @@
+        iconf->threads = 1;
+        SCReturnInt(0);
+    }
+
+    if (entry_str == NULL) {
+        SCLogError(SC_ERR_INVALID_VALUE, "Passed value is NULL");
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+
+    if (strcmp(entry_str, "auto") == 0) {
+        iconf->threads = (int)rte_lcore_count() - 1;
+        SCReturnInt(0);
+    }
+
+    if (StringParseInt32(&iconf->threads, 10, 0, entry_str) < 0) {
+        SCLogError(SC_ERR_INVALID_VALUE, "Value %s can not be parsed to int", entry_str);
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+
+    if (iconf->threads < 1 || iconf->threads > ((int)rte_lcore_count() - 1)) {
+        SCLogError(SC_ERR_INVALID_VALUE, "Invalid number of threads");
+        errno = ERANGE;
+        SCReturnInt(-errno);
+    }
+
+    SCReturnInt(0);
+}
+
+int ConfigSetRxQueues(DPDKIfaceConfig *iconf, uint16_t nb_queues)
+{
+    iconf->nb_rx_queues = nb_queues;
+    if (iconf->nb_rx_queues < 1) {
+        SCLogError(SC_ERR_INVALID_VALUE, "Number of rx queues must be positive");
+        errno = ERANGE;
+        SCReturnInt(-errno);
+    }
+
+    SCReturnInt(0);
+}
+
+int ConfigSetTxQueues(DPDKIfaceConfig *iconf, uint16_t nb_queues)
+{
+    iconf->nb_tx_queues = nb_queues;
+    if (iconf->nb_tx_queues < 1) {
+        SCLogError(SC_ERR_INVALID_VALUE, "Number of tx queues must be positive");
+        errno = ERANGE;
+        SCReturnInt(-errno);
+    }
+
+    SCReturnInt(0);
+}
+
+int ConfigSetMempoolSize(DPDKIfaceConfig *iconf, intmax_t entry_int)
+{
+    if (entry_int <= 0) {
+        SCLogError(SC_ERR_INVALID_VALUE, "Invalid memory pool size");
+        errno = ERANGE;
+        SCReturnInt(-errno);
+    }
+
+    iconf->mempool_size = entry_int;
+    SCReturnInt(0);
+}
+
+int ConfigSetMempoolCacheSize(DPDKIfaceConfig *iconf, intmax_t entry_int)
+{
+    if (entry_int <= 0 || entry_int > RTE_MEMPOOL_CACHE_MAX_SIZE) {
+        SCLogError(SC_ERR_INVALID_VALUE,
+                "Invalid memory pool cache size"
+                "(max %" PRIu32,
+                RTE_MEMPOOL_CACHE_MAX_SIZE);
+        errno = ERANGE;
+        SCReturnInt(-errno);
+    }
+
+    iconf->mempool_cache_size = entry_int;
+    SCReturnInt(0);
+}
+
+int ConfigSetRxDescriptors(DPDKIfaceConfig *iconf, intmax_t entry_int)
+{
+    if (entry_int <= 0) {
+        SCLogError(SC_ERR_DPDK_CONF, "Number of RX descriptors must be a positive number");
+        errno = ERANGE;
+        SCReturnInt(-errno);
+    }
+
+    iconf->nb_rx_desc = entry_int;
+    SCReturnInt(0);
+}
+
+int ConfigSetTxDescriptors(DPDKIfaceConfig *iconf, intmax_t entry_int)
+{
+    if (entry_int <= 0) {
+        SCLogError(SC_ERR_DPDK_CONF, "Number of TX descriptors must be a positive number");
+        errno = ERANGE;
+        SCReturnInt(-errno);
+    }
+
+    iconf->nb_rx_desc = entry_int;
+    SCReturnInt(0);
+}
+
+int ConfigSetMtu(DPDKIfaceConfig *iconf, intmax_t entry_int)
+{
+    if (entry_int < RTE_ETHER_MIN_MTU || entry_int > RTE_ETHER_MAX_JUMBO_FRAME_LEN) {
```

Comment:
does all NIC support Jumbo frame? Does all nic support segment for jumbo frame? Please fix this to be vendor neutral